pipeline_games/load_games.py

```diff
@@ -0,0 +1,256 @@
+    with conn.cursor(cursor_factory=RealDictCursor) as cur:
+        cur.execute(
+            """SELECT publisher_id FROM publisher WHERE publisher_name = %s;""",
+            [data[11]])
+        publisher_id = cur.fetchone()['publisher_id']
+        cur.execute(
+            """SELECT game_id FROM game WHERE app_id = %s;""",
+            [data[2]])
+        game_id = cur.fetchone()['game_id']
+        cur.execute(
+            """SELECT exists (SELECT 1 FROM game_publisher_link 
+            WHERE game_id = %s AND publisher_id = %s LIMIT 1);""",
+            [game_id, publisher_id])
+        result = cur.fetchone()
+        if result['exists'] is True:
+            cur.close()
+        elif result['exists'] is False:
+            cur.execute(
+                """INSERT INTO game_publisher_link(game_id, publisher_id)
+                VALUES (%s, %s);""", [game_id, publisher_id])
+        conn.commit()
+        cur.close()
+
+
+def add_to_developer_link_table(conn: connect, data: list) -> None:
+    """Update link table"""
+    with conn.cursor(cursor_factory=RealDictCursor) as cur:
+        cur.execute(
+            """SELECT developer_id FROM developer WHERE developer_name = %s;""",
+            [data[10]])
+        developer_id = cur.fetchone()['developer_id']
+        cur.execute(
+            """SELECT game_id FROM game WHERE app_id = %s;""",
+            [data[2]])
+        game_id = cur.fetchone()['game_id']
+        cur.execute(
+            """SELECT exists (SELECT 1 FROM game_developer_link 
+            WHERE game_id = %s AND developer_id = %s LIMIT 1);""",
+            [game_id, developer_id])
+        result = cur.fetchone()
+        if result['exists'] is True:
+            cur.close()
+        elif result['exists'] is False:
+            cur.execute(
+                """INSERT INTO game_developer_link(game_id, developer_id)
+                VALUES (%s, %s);""", [game_id, developer_id])
+        conn.commit()
+        cur.close()
+
+
+def find_platform_id_for_row(cache: dict, data: list) -> int:
+    """Returns platform id based on cache dict for platform"""
+    platforms_for_row = {'mac': data[8],
+                         'windows': data[7], 'linux': data[9]}
+    for key, value in cache.items():
+        if platforms_for_row.items() <= value.items():
+            return key
+        return None
+
+
+if __name__ == "__main__":
+    load_dotenv()
+    configuration = environ
+    connection = get_db_connection(configuration)
+
+    data_frame = pd.read_csv("genres.csv")
+    game_data = pd.read_csv("final_games.csv")
+
+    try:
+
+        developers_cache = {}
+        data_frame["developers"].apply(
+            lambda row: get_existing_data(
+                "developer_id", "developer", "developer_name",
+                row, connection, developers_cache))
+        new_developers = data_frame["developers"].apply(
+            lambda row: check_if_in_cache(row, developers_cache)).dropna(axis=0)
```

Comment:
So, from my understanding cache in here instead of improving execution time would actually do quite the opposite with the way it is applied. Happy to huddle about it